.github/workflows/build.yml

```diff
@@ -3,10 +3,10 @@ name: build
 on:
   push:
     branches:
-      - 'main'
```

Comment:
Normally the current behavior is OK: it runs when pushing to main, or on PRs to "main". I don't see the need for other cases, do you? If no you can revert (if it ain't broken don't fix it)

Reply:
I changed it so that CI runs tests when pushing on my `abstraction` branch (or any dev branch). I'll revert it !